File: src/main/java/co/herod/adbwrapper/AdbDeviceServices.java
package co.herod.adbwrapper;

import co.herod.adbwrapper.model.AdbDevice;
import org.jetbrains.annotations.Nullable;

import static co.herod.adbwrapper.util.Utils.enableOrDisabled;

@SuppressWarnings("unused")
public class AdbDeviceServices {

    private static final String SERVICE = "svc";

    private static final String SPACE = " ";

    public static void enableService(@Nullable final AdbDevice adbDevice, boolean enable, final String serviceType) {

        //noinspection StringBufferReplaceableByString
        Adb.INSTANCE.blocking(adbDevice, new StringBuilder()
                .append(AdbCommand.SHELL)
                .append(SPACE)
                .append(SERVICE)
                .append(SPACE)
                .append(serviceType)
                .append(SPACE)
                .append(enableOrDisabled(enable))
                .toString());
    }
}

File: adb-kotlin/src/main/java/co/herod/adbwrapper/AdbDeviceServices.kt
@file:Suppress("unused")

package co.herod.adbwrapper

import co.herod.adbwrapper.model.AdbDevice

fun AdbDevice.enableService(enable: Boolean, serviceType: String) {
    execute("${S.SHELL} ${S.SERVICE} $serviceType ${enable.toEnable()}")
}

private fun Boolean.toEnable() = when {
    this -> S.ENABLE
    else -> S.DISABLE
}